components/chassis.py

```diff
@@ -406,6 +406,16 @@ def is_stationary(self) -> bool:
             and math.isclose(velocity.omega, 0.0, abs_tol=math.radians(3))
         )
 
+    def stop(self) -> None:
+        for module in self.modules:
+            module.stop()
+            # Also reset the state to account for the internal smoothing
+            module.state = SwerveModuleState(0, module.get_rotation())
+        self.stop_snapping()
+        self.set_coast_in_neutral(coast_mode=False)
+        self.do_smooth = True
+        self.heading_controller.setPID(Kp=3.0, Ki=0.0, Kd=0.0)
```

Comment:
These should stay in `on_disable`, they're not related to stopping the motion of the chassis.